Show more per-camera telemetry on SmartDashboard

ivanggq opened the pull request:
- show confidence multiplier and estimated pose per camera on SmartDashboard
- shortened and organized the vision-related telemetry in the format `Vision/<camera name> - <description of data>`
- removed unused imports

ivanggq commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -9,7 +9,6 @@
 import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.CommandScheduler;
-import frc.robot.subsystems.swervedrive.SwerveSubsystem;
```
Removed a bunch of unused imports to reduce the issues shown in VS Code.

ivanggq commented on src/main/java/frc/robot/subsystems/vision/Vision.java:
```diff
@@ -28,13 +23,15 @@ public class Vision implements Runnable {
   private final PhotonPoseEstimator photonPoseEstimator;
   private final PhotonCamera photonCamera;
   private final AtomicReference<EstimatedRobotPose> atomicEstimatedRobotPose = new AtomicReference<EstimatedRobotPose>();
+  public final String cameraName;
```
Exposed the camera name from the `Vision` class.

ivanggq commented on src/main/java/swervelib/motors/SparkFlexSwerve.java:
```diff
@@ -363,7 +363,7 @@ public void set(double percentOutput)
   public void setReference(double setpoint, double feedforward)
   {
-    boolean possibleBurnOutIssue = true;
+//    boolean possibleBurnOutIssue = true;
```
Didn't want to delete this unused guys in case somebody is looking for them...

ivanggq commented on src/main/java/frc/robot/subsystems/vision/VisionEstimation.java:
```diff
@@ -157,26 +173,29 @@ private Matrix<N3, N1> confidenceCalculator(EstimatedRobotPose estimation) {
-    }
 
+        final String smartDashboardCamPoseKey = "Vision/" + estimator.cameraName + " - last pose";
+        SmartDashboard.putString(smartDashboardCamPoseKey, cameraPose != null ? cameraPose.estimatedPose.toString() : "no pose");
```
Reporting the pose from each camera separately so we can see what each camera thinks.

ivanggq commented on src/main/java/frc/robot/subsystems/vision/VisionEstimation.java:
```diff
@@ -157,26 +173,29 @@ private Matrix<N3, N1> confidenceCalculator(EstimatedRobotPose estimation) {
                         / (1
                                 + ((estimation.targetsUsed.size() - 1) * VisionConstants.TAG_PRESENCE_WEIGHT)));
 
+        confidenceTelemetry.addNumber(confidenceMultiplier);
```
Adding the current confidence multiplier to this `MeanPerPeriodTelemetry` helper, which will send the mean (avg) conf. mult. every half sec to SmartDashboard.

ivanggq commented on src/main/java/frc/robot/subsystems/vision/VisionEstimation.java:
```diff
@@ -110,6 +121,10 @@ public void periodic() {
+        // confidenceInkyTelemetry.periodic();
+        // confidenceBlinkyTelemetry.periodic();
+        confidencePinkyTelemetry.periodic();
+        confidenceClydeTelemetry.periodic();
```
The `periodic()` function of these helper telemetry classes will send the data to SmartDashboard once every sec or half sec (the reporting period is specified in their c'tor on lines 89-101).

Boomaa23 commented on src/main/java/frc/robot/subsystems/vision/Vision.java:
```diff
@@ -53,10 +50,8 @@ public Vision(PhotonCamera cameraName, Transform3d robotToCamera) {
-    setAtomicCountTelemetry = new CountPerPeriodTelemetry(
-        "Vision - " + cameraName.getName() + " - set atomic count per s",
-        1);
+    runCountTelemetry = new CountPerPeriodTelemetry("Vision/" + cameraName + " - runs per s", 1);
+    setAtomicCountTelemetry = new CountPerPeriodTelemetry("Vision/" + cameraName + " - meas per s push", 1);
```
Could use `String.format()` here if you want; not required though

ivanggq replied:
I think I will create a simple helper function that returns this telemetry string as this string/pattern is currently duplicated on 8+ places (vision/ + cameraname + telemetry name).

Boomaa23 commented on src/main/java/frc/robot/subsystems/vision/Vision.java:
```diff
@@ -69,17 +64,16 @@ public void run() {
         var photonResults = photonCamera.getLatestResult(); //Gets the latest camera results
 
-        // var test = VisionConstants.class.getDeclaredField("ROBOT_TO_" + photonCamera.getName().toUpperCase());
+        // var test = VisionConstants.class.getDeclaredField("ROBOT_TO_" + cameraName.toUpperCase());
         // test.setAccessible(true);
         // Transform3d test2 = (Transform3d) test.get(VisionConstants.class.getClass());
         //SmartDashboard.putString("Test", test2.toString());
```
Is there a reason these are here but commented? Remove if no reason

ivanggq replied:
Will remove. This was some code from @aayushext , not sure if we planned to use it.

Boomaa23 commented on src/main/java/frc/robot/subsystems/vision/VisionEstimation.java:
lots of commented things in this file. if you intend to use them move to another more relevant place, otherwise remove

ivanggq replied:
The telemetry lines/variables are usually in sets of 4 - one per camera. But since at the moment of this PR we had only 2 cameras, we added 2 uncommented lines and 2 commented lines. Eventually, all 4 cameras will be added and all code should be uncommented.

Boomaa23 commented on src/main/java/swervelib/motors/SparkMaxSwerve.java:
```diff
@@ -247,7 +247,7 @@ public void configurePIDF(PIDFConfig config)
 //    int pidSlot =
 //        isDriveMotor ? SparkMAX_slotIdx.Velocity.ordinal() : SparkMAX_slotIdx.Position.ordinal();
-    int pidSlot = 0;
+//    int pidSlot = 0;
```
wut dis

ivanggq replied:
Commented out unused variable, was unsure if it has sentimental value to the author who added it :) Might remove it now...

Boomaa23 reviewed (dismissed):
looks good to me content wise! again just minor refactoring, but honestly it'd be fine as is

ivanggq committed:
Remove unused imports.

ivanggq committed:
Rename SmartDashboard vision object names.

ivanggq committed:
Display confidence multiplier per camera on SmartDashboard.

ivanggq committed:
Show estimated pose of each camera on SmartDashboard.

ivanggq committed:
Added TelemUtils helper class.

ivanggq commented:
In the interest of time and in preparation for testing vision code today, I am going to merge this PR, dismissing your review @Boomaa23 . I think I addressed your comments but please let me know if you think more changes are needed and we can make changes in a different PR. Thanks.

ivanggq commented:
Hm, I cannot merge anyway without an approval, so will wait for an approval :) @Boomaa23 @AlexanderGardiner @aayushext @Coolkid0629
